core/src/main/java/org/testcontainers/containers/DockerComposeContainer.java

```diff
@@ -159,10 +161,14 @@ private void pullImages() {
         runWithCompose("pull");
     }
 
+    public SELF withServices(@NonNull String... services) {
+        this.services = Arrays.asList(services);
+        return self();
+    }
 
     private void createServices() {
         // Run the docker-compose container, which starts up the services
-        runWithCompose("up -d");
+        runWithCompose("up -d " + String.join(" ", this.services));
```

Comment:
In the case of `services` being an empty list, the run command will be `"up -d "`. _I know_ that for Docker Compose this is functionally identical and starts all services.

However, just to be explicit in our code, could we perhaps do an `if/else` check around this, and only concatenate the services list onto `"up -d"` if necessary?

I'm fine to have the `up` string duplicated in the code.

You might also notice that I'm being a hypocrit about explicitness of code, when the original `-d` here is itself not very clear 😁. If you wouldn't mind changing `-d` to `--detach` while you're editing the code, that would be another improvement.

Reply:
```suggestion
if(services.empty()) {
    runWithCompose("up -d");
} else {
    runWithCompose("up -d " + String.join(" ", services));
}
```
Would that be fine to you? Clearer separation is always better!

Note: The older versions of docker-compose do not support `--detach` (e.g. my  version of `1.17.1`). So that is why I would let it like that? Thanks for your suggestion :)

Reply:
That looks good to me - thanks.